client/src/hooks/useRequest.ts

```diff
@@ -0,0 +1,40 @@
+import type { RequestError, ValidationError, Fetcher, OnError, OnSuccess, OnLoading } from './../interface/ApiData';
+import { useState, useCallback, ReactElement } from 'react';
+import wait from '../helpers/wait';
+
+const useRequest = <D>(delay = 0) => {
```

Comment:
Maybe we can rename this hook as it may cause some confusion since we also have bookings/requests 😄